### PR TITLE
cli: improve help text

### PR DESCRIPTION
**Pull request by layday**

### CODE REVIEW

**Review by gaborbernat — APPROVED** [review]

File: src/build/__main__.py [hunk 1]
```diff
@@ -159,7 +158,8 @@ def main_parser():  # type: () -> argparse.ArgumentParser
         '--config-setting',
         '-C',
         action='append',
-        help='pass option to the backend',
+        help='pass options to the backend.  Options which begin with a hyphen must be in the form of '
+        '`--config-setting=--opt` or `-C--opt`',
```
Comment (FFY00) [inline src/build/__main__.py:162]: Maybe add the optional `=value` here?

File: src/build/__main__.py [hunk 2]
```diff
@@ -159,7 +158,8 @@ def main_parser():  # type: () -> argparse.ArgumentParser
         '--config-setting',
         '-C',
         action='append',
-        help='pass option to the backend',
+        help='pass options to the backend.  Options which begin with a hyphen must be in the form of '
```
[the remark this answers is not in the document]
Comment (layday) [reply src/build/__main__.py]: done

File: src/build/__main__.py (hunk 1 above)
Comment (layday) [reply src/build/__main__.py]: done

Comment (FFY00) [reply src/build/__main__.py]: Well, I was thinking `-C--opt(=value)`, sorry for not being clear :sweat_smile:

**Pull request merged**